os-server/src/main/java/app/openschool/auth/api/annotation/processor/SameValuesValidator.java

```diff
@@ -0,0 +1,45 @@
+package app.openschool.auth.api.annotation.processor;
+
+import app.openschool.auth.api.annotation.SameValues;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class SameValuesValidator implements ConstraintValidator<SameValues, Object> {
+
+  private List<String> fieldNames;
+
+  @Override
+  public void initialize(SameValues sameValues) {
+    fieldNames = Arrays.asList(sameValues.sameFields());
+  }
+
+  @Override
+  public boolean isValid(Object targetObject, ConstraintValidatorContext ctx) {
+    Set<Object> uniqueFieldValues =
+        fieldNames.stream()
+            .map(
+                fieldName -> {
+                  try {
```

Comment:
please move this lambda logic into a separate method, and create a new runtime exception for this case